src/bin/fish.rs

```diff
@@ -71,9 +71,99 @@
 
 const DOC_DIR: &str = env!("DOCDIR");
 const DATA_DIR: &str = env!("DATADIR");
+const DATA_DIR_SUBDIR: &str = env!("DATADIR_SUBDIR");
 const SYSCONF_DIR: &str = env!("SYSCONFDIR");
 const BIN_DIR: &str = env!("BINDIR");
 
+#[cfg(feature = "installable")]
+fn install(noconfirm: bool) {
+    use rust_embed::RustEmbed;
+
+    #[derive(RustEmbed)]
+    #[folder = "share/"]
+    struct Asset;
+
+    use std::fs;
+    use std::io::ErrorKind;
+    use std::io::Write;
+    use std::io::{stderr, stdin};
+    let (_, home) = fish::env::get_user_home();
+    let Some(home) = home else {
+        eprintln!("Can't find $HOME",);
+        std::process::exit(1);
+    };
+    let dir = PathBuf::from(home).join(DATA_DIR).join(DATA_DIR_SUBDIR);
+
+    // TODO: Translation,
+    // FLOG?
+    // - Install: Translations
+    // - Install: Manpages (build via build.rs)
+    // - Don't install: __fish_build_paths.fish.in
+    if !noconfirm {
+        eprintln!(
+            "This will write fish's data files to '{}'.\n\
+             Please enter 'yes' to continue.",
+            dir.display()
+        );
+        eprint!("> ");
+        let _ = stderr().flush();
+
+        let mut input = String::new();
+        stdin()
+            .read_line(&mut input)
+            .expect("Failed to read from stdin");
+
+        if input != "yes\n" {
+            eprintln!("Exiting without writing any files");
+            std::process::exit(1);
```

Comment:
absolute nit but when I use this noninteractive version, I get an abandoned
line (fish would fix it after printing ⏎ but bash doesn't)

	[johannes@localhost fish-shell]$ echo yes | fish --install
	This will write fish's data files to '/home/johannes/.local/share/fish/install'.
	Please enter 'yes' to continue.
	> [johannes@localhost fish-shell]$ 

I'm not yet sure what's the easiest fix (I guess we could reuse our normal
reader loop..).  I guess that's one upside of `fish --install=noconfirm`
in addition to making it more obvious to everyone how to do noninteractive installs.

Reply:
I made it skip the prompt if stdin isn't a tty.